app/src/androidTest/java/com/github/se/polyfit/data/remote/firebase/GeoQueryRepositoryTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.github.se.polyfit.data.remote.firebase
+
+import com.firebase.geofire.GeoFire
+import com.firebase.geofire.GeoLocation
+import com.firebase.geofire.GeoQuery
+import com.firebase.geofire.GeoQueryEventListener
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import io.mockk.Runs
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+
+class GeoQueryRepositoryTest {
+
+  private val mockRtdb: FirebaseDatabase = mockk()
+  private val mockGeoFireRef: DatabaseReference = mockk()
+  private lateinit var repository: PostFirebaseRepository
+
+  @Before
+  fun setUp() {
+    every { mockRtdb.getReference("posts_location") } returns mockGeoFireRef
+    every { mockGeoFireRef.orderByChild(any()) } returns mockGeoFireRef
+
+    every { mockGeoFireRef.orderByChild("someKey").equalTo("someValue" as String?) } returns
+        mockGeoFireRef
+    every { mockGeoFireRef.startAt(any<String>()) } returns mockGeoFireRef
+    every { mockGeoFireRef.endAt(any<String>()) } returns mockGeoFireRef
+
+    every { mockGeoFireRef.equalTo("someStringValue" as String?) } returns mockGeoFireRef
+    every { mockGeoFireRef.addChildEventListener(any()) } returns mockk()
+
+    every { mockGeoFireRef.addListenerForSingleValueEvent(any()) } just Runs
+
+    repository = PostFirebaseRepository(rtdb = mockRtdb)
+  }
+
+  @Test
+  fun queryNearbyPostsShouldTriggerGeoQueryEventListener() = runBlocking {
```

Comment:
Since runBlocking is an older function that has been replaced by runTest, it might be a good idea to use the latter in the future.